### PR TITLE
#18 Implement search filter for the color attribute

### PR DESCRIPTION
**Pull request by diazwatson**
### Description (*)

#18 Although it seems to be working there are two things I am not happy about it:
Screenshot 2019-06-14 at 01 51 16
Screenshot 2019-06-14 at 01 51 33

- Rules defined in `Magento_Ui:etc/ui_configuration.xsd` do not allow a colorPicker component to be added inside a listing component (should we extend `ui_configuration.xsd` to include this type of configuration?).
- As you can see, the label of the filter is aligned to the right, this is due to the class automatically added to the component (`admin__field-label`). the right class should be `admin__form-field-label`

I am going to continue reviewing this to see if I can fix the issues or see if I can come up with a better  way to implement this filter.

### CODE REVIEW

**Comment by diazwatson** [comment]
Hey Guys, I have refactored to use the template component class approach instead of the color picker but for some reason the label doesn't show. Has anyone seen similar issue with a Ui Component  before?

File: AdobeStockImageAdminUi/Ui/Component/Listing/Filter/Color.php [hunk 1]
```diff
@@ -0,0 +1,123 @@
+<?php
```
Comment (larsroettig) [inline AdobeStockImageAdminUi/Ui/Component/Listing/Filter/Color.php:1]: Please Use **strict_type** 

```php
<?php
/**
 *
 * Copyright © Magento, Inc. All rights reserved.
 * See COPYING.txt for license details.
 */
declare(strict_types=1);
```

File: AdobeStockImageAdminUi/Ui/Component/Listing/Filter/Color.php [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+use Magento\Framework\View\Element\UiComponent\ContextInterface;
+use Magento\Framework\View\Element\UiComponentFactory;
+use Magento\Ui\Component\Filters\FilterModifier;
+use Magento\Ui\Component\Form\Element\Input as ElementInput;
+use Magento\Ui\Component\Filters\Type\AbstractFilter;
+use Magento\Ui\Model\ColorPicker\ColorModesProvider;
+
+class Color extends AbstractFilter
+{
+    const NAME = 'filter_input';
```
Comment (larsroettig) [inline AdobeStockImageAdminUi/Ui/Component/Listing/Filter/Color.php:28]: why do you add their const for one-time usage? Usually, we have const only interfaces`

File: AdobeStockImageAdminUi/Ui/Component/Listing/Filter/Color.php [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+    const COMPONENT = 'input';
+
+    /**
+     * Wrapped component
+     *
+     * @var ElementInput
+     */
+    protected $wrappedComponent;
```
Comment (larsroettig) [inline AdobeStockImageAdminUi/Ui/Component/Listing/Filter/Color.php]: OPTIONAL: you should use private there to enforce setter and getter

File: AdobeStockImageAdminUi/Ui/Component/Listing/Filter/Color.php [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+    /**
+     * Provides color picker modes configuration
+     *
+     * @var ColorModesProvider
+     */
+    private $modesProvider;
+
+    public function __construct(
```
Comment (larsroettig) [inline AdobeStockImageAdminUi/Ui/Component/Listing/Filter/Color.php:54]: Missing doc block static should fail

File: AdobeStockImageAdminUi/Ui/Component/Listing/Filter/Color.php [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+        );
+
+        $this->applyFilter();
+
+        $this->initColorPickerConfig();
+
+        parent::prepare();
+    }
+
+    private function initColorPickerConfig()
```
Comment (larsroettig) [inline AdobeStockImageAdminUi/Ui/Component/Listing/Filter/Color.php]: Missing docblock what do

**Review by larsroettig — CHANGES_REQUESTED** [review]

**Comment by diazwatson** [comment]
Hi @larsroettig 
I believe all changes you requested are now covered, could you please review and let me know if any?

Cheers

**Review by sivaschenko — APPROVED** [review]

**Review by larsroettig — APPROVED** [review]

**Review by sidolov — APPROVED** [review]

**Pull request merged**